components/pedigree/rest/src/main/java/org/phenotips/pedigree/rest/internal/PedigreeConverterImpl.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+package org.phenotips.pedigree.rest.internal;
+
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.phenotips.pedigree.rest.PedigreeConverter;
+import org.xwiki.component.annotation.Component;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.slf4j.Logger;
+import org.xwiki.rest.XWikiResource;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Resource for converting PED files to JSON with REST
+ *
+ * @version $Id$
+ * @since 1.4m2
+ */
+@Component
+@Named("org.phenotips.pedigree.rest.internal.PedigreeConverterImpl")
+@Singleton
+public class PedigreeConverterImpl extends XWikiResource implements PedigreeConverter{
+    @Inject
+    private Logger logger;
+
+    @Override
+    public Response convertPedigree(String ped) {
+        if (ped == null) {
+            this.logger.error("PED file is empty");
```

Comment:
I contemplated a bit about asking to replace `error()`-s with `debug()`-s, but I think this (and the other case below)  ARE errors worth logging